lib/simplecov/filter.rb

```diff
@@ -63,11 +65,23 @@ def matches?(source_file)
   end
 
   class ArrayFilter < SimpleCov::Filter
-    # Returns true if any of the file paths passed in the given array matches the string
-    # configured when initializing this Filter with StringFilter.new(['some/path', 'other/path'])
+    def initialize(filter_argument)
+      filter_argument.map! do |arg|
+        if arg.is_a?(SimpleCov::Filter)
+          arg
+        else
+          Filter.class_for_argument(arg).new(arg)
+        end
+      end
+
+      super(filter_argument)
+    end
```

Comment:
I have a feeling, the additions here were necessary because THINGS weren't working. Sounds like we might miss some specs here. If you wanna add those, great! :+1: 

If not let me know and I'll try to get to it when I got some time :)